src/node_platform.cc

```diff
@@ -253,20 +257,22 @@ void PerIsolatePlatformData::PostIdleTaskImpl(
 
 void PerIsolatePlatformData::PostTaskImpl(std::unique_ptr<Task> task,
                                           const v8::SourceLocation& location) {
-  if (flush_tasks_ == nullptr) {
+  // Check if flush_tasks is valid before using it
+  if (!flush_tasks_valid_.load()) {
     // V8 may post tasks during Isolate disposal. In that case, the only
     // sensible path forward is to discard the task.
     return;
   }
-  foreground_tasks_.Push(std::move(task));
+  foreground_tasks_.Lock().Push(std::move(task));
   uv_async_send(flush_tasks_);
```

Comment:
I think what you want to do here is take out the lock, then check and push; then you don't need that atomic bool either. I can't post it as a suggestion but it'd look like this:
```c++
auto locked = foreground_tasks_.Lock();
if (flush_tasks == nullptr) return;
locked.Push(std::move(task));
uv_async_send(flush_tasks_);
```
The other thread must take out the lock when freeing `flush_tasks_`.